packages/typespec-python/src/emitter.ts

```diff
@@ -252,6 +226,17 @@ function getType(context: SdkContext, type: EmitterType): any {
     } else {
         newValue = emitType(context, type);
     }
+
+    if (oriType?.kind === "ModelProperty") {
+        if (context.program.checker.isStdType(oriType.type)) {
+            if (oriType.type.name === "utcDateTime" || oriType.type.name === "offsetDateTime") {
+                // if it's a date-time we change the format
+                newValue["format"] = isHeader(context.program, oriType) ? "date-time-rfc1123" : "date-time";
+            }
+        }
```

Comment:
For this logic, I copied from TCGC because I believe TCGC will remove `@clientFormat`, right? @iscai-msft

Reply:
we're not removing it yet, we will only remove it when we remove `@clientFormat`, so we're good to keep the code, let me update the pr. thanks!

Reply:
I've added back `getClientFormat`, please help to review.